packages/frontend/pages/lobby.tsx

```diff
@@ -1,23 +1,35 @@
-import { ReactElement } from 'react';
+import { ReactElement, useEffect } from 'react';
 import { connect } from 'react-redux';
 import { Grid } from 'semantic-ui-react';
 import Link from 'next/link';
 
 import { RootState } from '../store/index';
 import { ThemeOptions, IUserOptionsState } from '../store/userOptions/types';
+import { WindowProps, IWindowPropsState, IGetWidthAction } from '../store/system/types';
+import getWindowWidth from '../store/system/actions';
 
 import Nav from '../components/nav';
 import OnlinePlayers from '../components/lobby/onlinePlayers';
 import Taako from '../components/lobby/taako';
 import Chat from '../components/lobby/chat';
 import GamesMenu from '../components/lobby/gamesMenu/gamesMenu';
+import Announcements from '../components/lobby/announcements';
 
 interface IProps {
   theme: ThemeOptions;
+  windowProps: WindowProps;
+  dispatchGetWindowWidth: typeof getWindowWidth;
 }
 
 const Lobby = (props: IProps): ReactElement => {
-  const { theme } = props;
+  const { theme, windowProps, dispatchGetWindowWidth } = props;
+  
+  useEffect(() => {
+    const resizeWindow = (): IGetWidthAction => dispatchGetWindowWidth(window.innerWidth);
+    window.addEventListener('resize', resizeWindow);
```

Comment:
I remember that you had to remove the previous listener whenever useEffect was called. Otherwise every frame update during resize, there'd be more and more eventlisteners added on? Could you check this pls?

Reply:
I changed it to
`  const resizeWindow = (): IGetWidthAction => dispatchGetWindowWidth(window.innerWidth);

  useEffect(() => {
    window.addEventListener('resize', resizeWindow);
    return (): void => window.removeEventListener('resize', resizeWindow);
  }, ['resize', resizeWindow])`

Reply:
Perfect, that looks good!